From 9cd794c57af3cbe89b802e1a94a0c3f16530fefc Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Tue, 4 Jan 2011 15:52:26 +0800
Subject: [PATCH] ENGR00137558-4 GPMI : remove the imx23-specific hooks

The check_transcription_stamp() and write_transcription_stamp() are
imx23-specific hooks.

Change them to local functions, and remove the hooks in structure
boot_rom_helper{}.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v0.c |    8 ++------
 drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v1.c |    4 +---
 drivers/mtd/nand/gpmi-nfc/gpmi-nfc.h        |    6 ------
 3 files changed, 3 insertions(+), 15 deletions(-)

diff --git a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v0.c b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v0.c
index 4212dbe..83c2208 100644
--- a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v0.c
+++ b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v0.c
@@ -288,7 +288,6 @@ static int imx23_rom_extra_init(struct gpmi_nfc_data  *this)
 {
 	struct device             *dev      =  this->dev;
 	struct physical_geometry  *physical = &this->physical_geometry;
-	struct boot_rom_helper    *rom      =  this->rom;
 	struct mil                *mil      = &this->mil;
 	struct nand_chip          *nand     = &mil->nand;
 	struct mtd_info           *mtd      = &mil->mtd;
@@ -306,7 +305,7 @@ static int imx23_rom_extra_init(struct gpmi_nfc_data  *this)
 	 * transcription stamp. If we find it, then we don't have to do
 	 * anything -- the block marks are already transcribed.
 	 */
-	if (rom->check_transcription_stamp(this))
+	if (check_transcription_stamp(this))
 		return 0;
 
 	/*
@@ -373,12 +372,11 @@ static int imx23_rom_extra_init(struct gpmi_nfc_data  *this)
 	}
 
 	/* Write the stamp that indicates we've transcribed the block marks. */
-	rom->write_transcription_stamp(this);
+	write_transcription_stamp(this);
 	return 0;
 }
 
 /* This structure represents the Boot ROM Helper for this version. */
-
 struct boot_rom_helper  gpmi_nfc_boot_rom_helper_v0 = {
 	.version                   = 0,
 	.description               = "Single/dual-chip boot area, "
@@ -386,6 +384,4 @@ struct boot_rom_helper  gpmi_nfc_boot_rom_helper_v0 = {
 	.swap_block_mark           = false,
 	.set_geometry              = set_geometry,
 	.rom_extra_init		   = imx23_rom_extra_init,
-	.check_transcription_stamp = check_transcription_stamp,
-	.write_transcription_stamp = write_transcription_stamp,
 };
diff --git a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v1.c b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v1.c
index 49cb329..699dea8 100644
--- a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v1.c
+++ b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-rom-v1.c
@@ -1,7 +1,7 @@
 /*
  * Freescale GPMI NFC NAND Flash Driver
  *
- * Copyright (C) 2010 Freescale Semiconductor, Inc.
+ * Copyright (C) 2010-2011 Freescale Semiconductor, Inc.
  * Copyright (C) 2008 Embedded Alley Solutions, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
@@ -77,6 +77,4 @@ struct boot_rom_helper  gpmi_nfc_boot_rom_helper_v1 = {
 						"block mark swapping supported",
 	.swap_block_mark           = true,
 	.set_geometry              = set_geometry,
-	.check_transcription_stamp = 0,
-	.write_transcription_stamp = 0,
 };
diff --git a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc.h b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc.h
index f76f4b5..e76700c 100644
--- a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc.h
+++ b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc.h
@@ -561,10 +561,6 @@ struct nfc_hal {
  *                              mark with the first byte of the OOB.
  * @set_geometry:               Configures the Boot ROM geometry.
  * @rom_extra_init:             Arch-specific init.
- * @check_transcription_stamp:  Checks for a transcription stamp. This pointer
- *                              is ignored if swap_block_mark is set.
- * @write_transcription_stamp:  Writes a transcription stamp. This pointer
- *                              is ignored if swap_block_mark is set.
  */
 
 struct boot_rom_helper {
@@ -573,8 +569,6 @@ struct boot_rom_helper {
 	const int           swap_block_mark;
 	int  (*set_geometry)             (struct gpmi_nfc_data *);
 	int  (*rom_extra_init)           (struct gpmi_nfc_data *);
-	int  (*check_transcription_stamp)(struct gpmi_nfc_data *);
-	int  (*write_transcription_stamp)(struct gpmi_nfc_data *);
 };
 
 /*
-- 
1.5.4.4

